From ba73a468b00842c927b8e3f773b8cfb339674620 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Tue, 8 Jan 2013 20:53:56 +0200
Subject: [PATCH 043/184] lcd3-cape: Change into using the lcdc DRM driver
 instead of da8xx-fb

Use the new DRM driver instead of the da8xx-fb driver.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 firmware/capes/cape-bone-lcd3-00A0.dts |   65 ++++++++++++++++++++++++++++----
 1 file changed, 58 insertions(+), 7 deletions(-)

diff --git a/firmware/capes/cape-bone-lcd3-00A0.dts b/firmware/capes/cape-bone-lcd3-00A0.dts
index 0a3079e..83a6d88 100644
--- a/firmware/capes/cape-bone-lcd3-00A0.dts
+++ b/firmware/capes/cape-bone-lcd3-00A0.dts
@@ -79,8 +79,13 @@
 
 	fragment@2 {
 		target = <&ocp>;
+
 		__overlay__ {
 
+			/* avoid stupid warning */
+			#address-cells = <1>;
+			#size-cells = <1>;
+
 			tscadc {
 				compatible = "ti-tscadc-dt";
 
@@ -161,18 +166,64 @@
 				};
 			};
 
-			lcd3-cape {
-				compatible = "da8xx-dt";
-
+//			lcd3-cape {
+//				compatible = "da8xx-dt";
+//
+//				pinctrl-names = "default";
+//				pinctrl-0 = <&bone_lcd3_cape_lcd_pins>;
+//
+//				ti,hwmods = "lcdc";
+//
+//				disp-pll = <16000000>;
+//				panel-type = "CDTech_S035Q01";
+//
+//			};
+
+			/* Settings for CDTech_S035Q01 / LCD3 cape: */
+			panel {
+				compatible = "lcdc,panel";
 				pinctrl-names = "default";
 				pinctrl-0 = <&bone_lcd3_cape_lcd_pins>;
+				panel-info {
+					ac-bias           = <255>;
+					ac-bias-intrpt    = <0>;
+					dma-burst-sz      = <16>;
+					bpp               = <16>;
+					fdd               = <0x80>;
+					tft-alt-mode      = <0>;
+					stn-565-mode      = <0>;
+					mono-8bit-mode    = <0>;
+					invert-line-clock = <1>;
+					invert-frm-clock  = <1>;
+					sync-edge         = <0>;
+					sync-ctrl         = <1>;
+					raster-order      = <0>;
+					fifo-th           = <0>;
+				};
+				display-timings {
+					native-mode = <&timing0>;
+					timing0: 320x240 {
+						hactive         = <320>;
+						vactive         = <240>;
+						hback-porch     = <21>;
+						hfront-porch    = <58>;
+						hsync-len       = <47>;
+						vback-porch     = <11>;
+						vfront-porch    = <23>;
+						vsync-len       = <2>;
+						clock-frequency = <8000000>;
+					};
+				};
+			};
 
+			fb {
+				compatible = "ti,am33xx-lcdc";
+				reg = <0x4830e000 0x1000>;
+				interrupt-parent = <&intc>;
+				interrupts = <36>;
 				ti,hwmods = "lcdc";
-
-				disp-pll = <16000000>;
-				panel-type = "CDTech_S035Q01";
-
 			};
+
 		};
 	};
 };
-- 
1.7.10.4

